Models/ProjectModels/Admin/AdminModel.php

```diff
@@ -0,0 +1,218 @@
+        }
+
+        return self::CONDITION_DATA[$actionName];
+    }
+
+    /**
+     * @param string $actionName
+     * @return string[]
+     */
+    protected function getDbFields(string $actionName): array
+    {
+        if (!array_key_exists($actionName, self::DB_FIELDS)) {
+            throw new \InvalidArgumentException('You forgot to add field with value in const DB_FIELDS!');
+        }
+
+        return self::DB_FIELDS[$actionName];
+    }
+
+    /**
+     * @param array $data
+     * @param string $fieldName
+     * @throws \Exception
+     */
+    public function changeAccess(array $data, string $fieldName): void
+    {
+        $count = count($data);
+        if ($fieldName === 'is_head' && $count > 1) {
+            $this->msgModel->setMsg('failure', 'too_much_admins', 'too_much_admins');
+
+            return;
+        }
+        
+        $conditionData = $this->formatArrayData(['id'], $data);
+        $this->setDbMsgModel();
+        $dbResult = $this->db->select(['id'])->from(['admins'])->condition($conditionData)->query()->fetchAll();
+        if (!$dbResult) {
+            throw new \Exception(
+                'Wrong Admin id:' . $data['id'] . ', during changeAccess->selectAdmins methods!'
+            );
+        }
+
+        $updateData = $this->formatArrayData([$fieldName], $data);
+        $dbResult = $this->db->update(['admins'], $updateData)->condition($conditionData)->exec();
+        if ($dbResult) {
+            $this->msgModel->setMsg('success', $this->getRefererAction(), 'success');
+            if ($fieldName === 'is_head') {
+                $conditionData = $this->formatArrayData(['id'], $this->sessionModel->getUserData());
+                $dbResult = $this->db->update(['admins'], ['is_head' => 0])->condition($conditionData)->exec();
+                if ($dbResult) {
+                    $this->sessionModel->setUserData(['is_head' => '0']);
+                } else {
+                    $this->logger->log(
+                        'default',
+                        'Problems to update db data of previous HeadAdmin '.
+                        $this->sessionModel->getUserData()['name'] . ':(' .
+                        $this->sessionModel->getUserData()['login'] . ')!'
+                    );
+                }
+            }
+        } else {
+            throw new \Exception(
+                'Something went wrong during updating Admin DB data,
+                 using : ' . $this->getRequestAction() . 'Action->changeAccess methods!'
+            );
+        }
+    }
+
+    /**
+     * @param array $fieldNames
+     * @param array $data
+     * @return array
+     */
+    private function formatArrayData(array $fieldNames, array $data): array
+    {
+        $result = [];
+        foreach ($fieldNames as $fieldName) {
+            foreach ($data as $field => $value) {
+                if (is_array($value)) {
+                    if (array_key_exists($fieldName, $value)) {
```

Comment:
why can't you unite is_array and array_key_exists inside one if condition? 
and after that check the rest inside the elseif ? 

```
private function formatArrayData(array $fieldNames, array $data): array
{
    $result = [];

    foreach ($fieldNames as $fieldName) {
        foreach ($data as $field => $value) {
            if (is_array($value) && array_key_exists($fieldName, $value)) {
                if (!isset($result[$fieldName])) {
                    $result[$fieldName] = [];
                }
                if (!in_array($value[$fieldName], $result[$fieldName])) {
                    $result[$fieldName][] = $value[$fieldName];
                }
            } elseif ($field === $fieldName) {
                $result[$fieldName] = $value;
            }
        }
    }

    if (empty($result)) {
        throw new InvalidArgumentException('Cannot find field in provided array data!');
    } elseif (count($result) === 1) {
        foreach ($result as $field => $value) {
            if (is_array($value) && count($value) === 1) {
                $result[$field] = $value[0];
            }
        }
    }

    return $result;
}
```

Please double check does it work the same as you wanted previously.

Reply:
looks like works well, the same as i needed, but have less lines with code. 
i will use it .
also i spent like 2-3 hours to create my method formatArrayData().  u r really like rabbit energizer, i dont know how u wrote so fast your vision if this, really u did this very fast
GJPLHF